hubedit/src/main.rs

```diff
@@ -159,6 +168,42 @@ fn main() -> Result<()> {
             archive.erase_caboose()?;
             archive.overwrite()?;
         }
+        Command::ReadVersion => {
```

Comment:
The semantics of this subcommand are confusing.  It reads both version and epoch from the caboose, parses them _very generously_ (defaulting to "" / 0 even if they exist but are unparseable), prints both of them, and then returns with an error if the image header epoch is different from the caboose epoch.

I'd suggest splitting it into two or three subcommands, something like `ReadVersion`, `ReadEpoch`, `CheckEpoch` (or have a `--check` flag in `ReadEpoch`).

I would also like for them to be much more strict: if you call `hubedit read-epoch` and the epoch either doesn't exist or is unparseable as an integer, it should return an error.  Same for `read-version`, etc.

Reply:
This is left over from when I had both ImageHeader and Caboose values that needed to be checked against each other. It isn't useful anymore and has been removed.